### DISCUSSION TITLE
Keep File System Routing when changing the pages folder name

### DISCUSSION THREAD

**Discussion by phiberber** [vikejs/vike#884] [Discussion] [answered]
I'm thinking of changing the pages folder name while using the V1 Design. Brillout once mentioned that VPS does not care about the name of the folder, but how does one accomplish File System Routing through a custom folder name? The only way I found was using `+route.ts` files, which in my opinion is a compromise when forced to do it for every route.

I don't think making it through extensions would be the right way to do it.

**Comment by brillout** [comment]
You can rename `/pages/` to `/whatever-name/` and it will work exactly the same, see https://vite-plugin-ssr.com/filesystem-routing.

**Reply by phiberber** [reply]
Actually, I've just noticed what's happening. Seems that when I change the pages folder name to something like app, the routes become localhost:3000/app. Which again, can be solved through Route Strings, though it doesn't seem to work exactly the same.

**Reply by brillout** [reply]
Further down below:

> We can change the URL root of Filesystem Routing by using _default.page.route.js#filesystemRoutingRoot.

You may want to read the docs more thoroughly before asking :-).

**Reply by phiberber** [reply]
Actually, I've read that part, but in my head didn't think that would be a solution to the problem as I attempted it before but it didn't work, guess I did something wrong, thanks!

**Reply by phiberber** [reply]
It would be nice to have some info on the V1 Design about the filesystemRoutingRoot property I guess.

**Reply by phiberber** [reply]
Actually, now I've seen this: https://github.com/brillout/vite-plugin-ssr/blob/79357062a3bd2684a8e7ae999f22eb3dcf516823/vite-plugin-ssr/node/plugin/plugins/importUserCode/v1-design/TODO#L10
Does filesystemRoutingRoot work with v1?

**Reply by brillout** [reply, marked as answer]
https://github.com/brillout/vite-plugin-ssr/issues/578#issuecomment-1549270183. ETA this week.
